### ISSUE TITLE
[Consensus/beacon] Failed to handle event log error with Lighthouse v4.2.0

### ISSUE THREAD

**Issue by jmc000** [ethpandaops/ethereum-metrics-exporter#83]
Hey, I get error logs for the consensus module that seems strange to me:

## Versions

- distro: `Ubuntu 20.04 LTS`
- ethereum-metrics-exporter: `0.21.0`
- execution client: `erigon/2.45.1/linux-amd64/go1.19.3`
- consensus client: `Lighthouse/v4.2.0-c547a11/x86_64-linux`
- network: `goerli`

## Problem
When starting the `ethereum-metrics-exporter` binary I got these errors in the log while the exporter seems to be working well:

```bash
{"component":"exporter","level":"error","module":"consensus/beacon","msg":"Failed to handle event: invalid voluntary exit event","time":"2023-06-21T18:09:05+02:00"}
```

## Launch command

```bash
ethereum-metrics-exporter \
  --metrics-port 1234 \
  --consensus-url http://localhost:5052 \
  --execution-url http://localhost:8545 \
  --monitored-directories /home/ethereum/var/lib
```

## Logs 
```bash
Started Ethereum exporter service.
{"cfgFile":"","level":"info","msg":"Loading config","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","msg":"Initializing...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","modules":"eth, net, web3","msg":"Initializing execution...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Enabling sync status metrics","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Enabling general metrics","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Enabling block metrics","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Enabling web3 metrics","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Enabling net metrics","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","msg":"Initializing disk usage...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","consensus_url":"http://localhost:5052","execution_url":"http://localhost:8545","level":"info","msg":"Starting metrics server on :1234","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","execution_url":"http://localhost:8545","level":"info","msg":"Starting execution metrics...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","msg":"Starting disk usage metrics...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","consensus_url":"http://localhost:5052","level":"info","msg":"Starting consensus metrics...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","exporter":"execution","level":"info","msg":"Started metrics exporter jobs","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","module":"consensus/beacon","msg":"Initializing beacon state","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"info","module":"consensus/beacon","msg":"Beacon state initialized! Ready to serve requests...","time":"2023-06-21T17:55:25+02:00"}
{"component":"exporter","level":"error","module":"consensus/beacon","msg":"Failed to handle event: invalid voluntary exit event","time":"2023-06-21T17:59:42+02:00"}
{"component":"exporter","level":"error","module":"consensus/beacon","msg":"Failed to handle event: invalid voluntary exit event","time":"2023-06-21T18:00:43+02:00"}
{"component":"exporter","level":"error","module":"consensus/beacon","msg":"Failed to handle event: invalid voluntary exit event","time":"2023-06-21T18:01:44+02:00"}
{"component":"exporter","level":"error","module":"consensus/beacon","msg":"Failed to handle event: invalid voluntary exit event","time":"2023-06-21T18:02:45+02:00"}
```

## Checking consensus client HTTP port
The beacon chain is well responding on the 5052 port, with no error logs. Same for the execution client, my validator is well attesting on goerli.

```bash
$ curl -s -X GET "http://localhost:5052/eth/v1/beacon/headers/head" -H  "accept: application/json" | jq
{
  "execution_optimistic": false,
  "finalized": false,
  "data": {
    "root": "0x233a05580d58acde8de50a761727e787f5f5fd26b77b90b190e83b44bb038611",
    "canonical": true,
    "header": {
      "message": {
        "slot": "5904592",
        "proposer_index": "7141",
        "parent_root": "0x38f27cf545ce8b819f7836dd9f403de6cc97f21aaf43a5ad381f9e94e39e488c",
        "state_root": "0x4be4a6228ca9ce5e6361978f23bbb7a0f7279e82bc93d5cf03a1cc8690a7cb93",
        "body_root": "0xced47a8b6bb0b79fe1f027966630695cc97f2fba1488881c867132f8d320228d"
      },
      "signature": "0xa4242f45a124ad248c3a0c9df7f78feea0d99d607b88905435eb21bc9ba3ad9c6945db9b61406763903c4a337d11be9517032269a565d8eceb573fa2a59fce7317de933be188de711b8edd6a4e6c2dc4b6698ee0a015bf3eaa8949b29fa567cd"
    }
  }
}
```

## Ethereum exporter Metrics 
```bash
# HELP eth_con_beacon_attestations The amount of attestations in the block.
# TYPE eth_con_beacon_attestations gauge
eth_con_beacon_attestations{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 128
eth_con_beacon_attestations{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 128
# HELP eth_con_beacon_deposits The amount of deposits in the block.
# TYPE eth_con_beacon_deposits gauge
eth_con_beacon_deposits{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 0
eth_con_beacon_deposits{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 0
# HELP eth_con_beacon_empty_slots_count The number of slots that have expired without a block proposed.
# TYPE eth_con_beacon_empty_slots_count counter
eth_con_beacon_empty_slots_count{module="beacon",node="consensus"} 0
# HELP eth_con_beacon_finality_checkpoint_epochs That epochs of the finality checkpoints.
# TYPE eth_con_beacon_finality_checkpoint_epochs gauge
eth_con_beacon_finality_checkpoint_epochs{checkpoint="finalized",module="beacon",node="consensus",state_id="head"} 184517
eth_con_beacon_finality_checkpoint_epochs{checkpoint="justified",module="beacon",node="consensus",state_id="head"} 184518
eth_con_beacon_finality_checkpoint_epochs{checkpoint="previous_justified",module="beacon",node="consensus",state_id="head"} 184517
# HELP eth_con_beacon_proposer_delay The delay of the proposer.
# TYPE eth_con_beacon_proposer_delay histogram
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="0"} 0
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="1000"} 15
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="2000"} 56
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="3000"} 62
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="4000"} 64
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="5000"} 64
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="6000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="7000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="8000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="9000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="10000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="11000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="12000"} 65
eth_con_beacon_proposer_delay_bucket{module="beacon",node="consensus",le="+Inf"} 65
eth_con_beacon_proposer_delay_sum{module="beacon",node="consensus"} 93345
eth_con_beacon_proposer_delay_count{module="beacon",node="consensus"} 65
# HELP eth_con_beacon_reorg_count The count of reorgs.
# TYPE eth_con_beacon_reorg_count counter
eth_con_beacon_reorg_count{module="beacon",node="consensus"} 0
# HELP eth_con_beacon_reorg_depth The number of reorgs.
# TYPE eth_con_beacon_reorg_depth counter
eth_con_beacon_reorg_depth{module="beacon",node="consensus"} 0
# HELP eth_con_beacon_slashings The amount of slashings in the block.
# TYPE eth_con_beacon_slashings gauge
eth_con_beacon_slashings{block_id="finalized",module="beacon",node="consensus",type="attester",version="CAPELLA"} 0
eth_con_beacon_slashings{block_id="finalized",module="beacon",node="consensus",type="proposer",version="CAPELLA"} 0
eth_con_beacon_slashings{block_id="head",module="beacon",node="consensus",type="attester",version="CAPELLA"} 0
eth_con_beacon_slashings{block_id="head",module="beacon",node="consensus",type="proposer",version="CAPELLA"} 0
# HELP eth_con_beacon_slot The slot number in the block.
# TYPE eth_con_beacon_slot gauge
eth_con_beacon_slot{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 5.904575e+06
eth_con_beacon_slot{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 5.904654e+06
# HELP eth_con_beacon_transactions The amount of transactions in the block.
# TYPE eth_con_beacon_transactions gauge
eth_con_beacon_transactions{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 79
eth_con_beacon_transactions{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 157
# HELP eth_con_beacon_voluntary_exits The amount of voluntary exits in the block.
# TYPE eth_con_beacon_voluntary_exits gauge
eth_con_beacon_voluntary_exits{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 0
eth_con_beacon_voluntary_exits{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 0
# HELP eth_con_beacon_withdrawals The amount of withdrawals in the block.
# TYPE eth_con_beacon_withdrawals gauge
eth_con_beacon_withdrawals{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 16
eth_con_beacon_withdrawals{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 16
# HELP eth_con_beacon_withdrawals_amount_gwei The sum amount of all the withdrawals in the block (in gwei).
# TYPE eth_con_beacon_withdrawals_amount_gwei gauge
eth_con_beacon_withdrawals_amount_gwei{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 3.2764565e+07
eth_con_beacon_withdrawals_amount_gwei{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 3.3202973e+07
# HELP eth_con_beacon_withdrawals_index_max The maximum index of the withdrawals in the block.
# TYPE eth_con_beacon_withdrawals_index_max gauge
eth_con_beacon_withdrawals_index_max{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 8.944379e+06
eth_con_beacon_withdrawals_index_max{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 8.945435e+06
# HELP eth_con_beacon_withdrawals_index_min The minimum index of the withdrawals in the block.
# TYPE eth_con_beacon_withdrawals_index_min gauge
eth_con_beacon_withdrawals_index_min{block_id="finalized",module="beacon",node="consensus",version="CAPELLA"} 8.944364e+06
eth_con_beacon_withdrawals_index_min{block_id="head",module="beacon",node="consensus",version="CAPELLA"} 8.94542e+06
# HELP eth_con_event_count The count of beacon events.
# TYPE eth_con_event_count counter
eth_con_event_count{event="attestation",module="event",node="consensus"} 38392
eth_con_event_count{event="block",module="event",node="consensus"} 65
eth_con_event_count{event="contribution_and_proof",module="event",node="consensus"} 934
eth_con_event_count{event="finalized_checkpoint",module="event",node="consensus"} 2
eth_con_event_count{event="head",module="event",node="consensus"} 65
eth_con_event_count{event="voluntary_exit",module="event",node="consensus"} 14
# HELP eth_con_event_time_since_last_subscription_event_ms The amount of time since the last subscription event (in milliseconds).
# TYPE eth_con_event_time_since_last_subscription_event_ms gauge
eth_con_event_time_since_last_subscription_event_ms{module="event",node="consensus"} 0
# HELP eth_con_fork_activated The activation status of the fork (1 for activated).
# TYPE eth_con_fork_activated gauge
eth_con_fork_activated{fork="ALTAIR",module="fork",node="consensus"} 1
eth_con_fork_activated{fork="BELLATRIX",module="fork",node="consensus"} 1
eth_con_fork_activated{fork="CAPELLA",module="fork",node="consensus"} 1
eth_con_fork_activated{fork="GENESIS",module="fork",node="consensus"} 1
# HELP eth_con_fork_current The current fork.
# TYPE eth_con_fork_current gauge
eth_con_fork_current{fork="CAPELLA",module="fork",node="consensus"} 1
# HELP eth_con_fork_epoch The epoch for the fork.
# TYPE eth_con_fork_epoch gauge
eth_con_fork_epoch{fork="ALTAIR",module="fork",node="consensus"} 36660
eth_con_fork_epoch{fork="BELLATRIX",module="fork",node="consensus"} 112260
eth_con_fork_epoch{fork="CAPELLA",module="fork",node="consensus"} 162304
eth_con_fork_epoch{fork="GENESIS",module="fork",node="consensus"} 0
# HELP eth_con_health_check_results_total Total of health checks results.
# TYPE eth_con_health_check_results_total counter
eth_con_health_check_results_total{module="health",node="consensus",result="success"} 62
# HELP eth_con_health_up Whether the node is up or not.
# TYPE eth_con_health_up gauge
eth_con_health_up{module="health",node="consensus"} 1
# HELP eth_con_node_version The version of the running beacon node.
# TYPE eth_con_node_version gauge
eth_con_node_version{module="general",node="consensus",version="Lighthouse/v4.2.0-c547a11/x86_64-linux"} 1
# HELP eth_con_peers The count of peers connected to beacon node.
# TYPE eth_con_peers gauge
eth_con_peers{direction="inbound",module="general",node="consensus",state="connected"} 24
eth_con_peers{direction="inbound",module="general",node="consensus",state="connecting"} 0
eth_con_peers{direction="inbound",module="general",node="consensus",state="disconnected"} 61
eth_con_peers{direction="inbound",module="general",node="consensus",state="disconnecting"} 0
eth_con_peers{direction="outbound",module="general",node="consensus",state="connected"} 76
eth_con_peers{direction="outbound",module="general",node="consensus",state="connecting"} 0
eth_con_peers{direction="outbound",module="general",node="consensus",state="disconnected"} 180
eth_con_peers{direction="outbound",module="general",node="consensus",state="disconnecting"} 0
# HELP eth_con_spec_base_reward_factor The base reward factor.
# TYPE eth_con_spec_base_reward_factor gauge
eth_con_spec_base_reward_factor{module="spec",node="consensus"} 64
# HELP eth_con_spec_config_name The name of the config.
# TYPE eth_con_spec_config_name gauge
eth_con_spec_config_name{module="spec",name="prater",node="consensus"} 1
# HELP eth_con_spec_deposit_chain_id The chain ID of the deposit contract.
# TYPE eth_con_spec_deposit_chain_id gauge
eth_con_spec_deposit_chain_id{module="spec",node="consensus"} 5
# HELP eth_con_spec_effective_balance_increment The effective balance increment.
# TYPE eth_con_spec_effective_balance_increment gauge
eth_con_spec_effective_balance_increment{module="spec",node="consensus"} 1e+09
# HELP eth_con_spec_epochs_per_sync_committee_period The number of epochs per sync committee period.
# TYPE eth_con_spec_epochs_per_sync_committee_period gauge
eth_con_spec_epochs_per_sync_committee_period{module="spec",node="consensus"} 256
# HELP eth_con_spec_eth1_follow_distance The number of blocks to follow behind the head of the eth1 chain.
# TYPE eth_con_spec_eth1_follow_distance gauge
eth_con_spec_eth1_follow_distance{module="spec",node="consensus"} 2048
# HELP eth_con_spec_genesis_delay The number of epochs to wait before processing the genesis block.
# TYPE eth_con_spec_genesis_delay gauge
eth_con_spec_genesis_delay{module="spec",node="consensus"} 1.919188e+06
# HELP eth_con_spec_max_attestations The maximum number of attestations.
# TYPE eth_con_spec_max_attestations gauge
eth_con_spec_max_attestations{module="spec",node="consensus"} 128
# HELP eth_con_spec_max_deposits The maximum number of deposits.
# TYPE eth_con_spec_max_deposits gauge
eth_con_spec_max_deposits{module="spec",node="consensus"} 16
# HELP eth_con_spec_max_effective_balance The maximum effective balance.
# TYPE eth_con_spec_max_effective_balance gauge
eth_con_spec_max_effective_balance{module="spec",node="consensus"} 3.2e+10
# HELP eth_con_spec_max_validators_per_committee The maximum number of validators per committee.
# TYPE eth_con_spec_max_validators_per_committee gauge
eth_con_spec_max_validators_per_committee{module="spec",node="consensus"} 2048
# HELP eth_con_spec_min_deposit_amount The minimum deposit amount.
# TYPE eth_con_spec_min_deposit_amount gauge
eth_con_spec_min_deposit_amount{module="spec",node="consensus"} 1e+09
# HELP eth_con_spec_min_genesis_active_validator_count The minimum number of active validators at genesis.
# TYPE eth_con_spec_min_genesis_active_validator_count gauge
eth_con_spec_min_genesis_active_validator_count{module="spec",node="consensus"} 16384
# HELP eth_con_spec_min_sync_committee_participants The minimum number of sync committee participants.
# TYPE eth_con_spec_min_sync_committee_participants gauge
eth_con_spec_min_sync_committee_participants{module="spec",node="consensus"} 1
# HELP eth_con_spec_preset_base The base of the preset.
# TYPE eth_con_spec_preset_base gauge
eth_con_spec_preset_base{module="spec",node="consensus",preset="mainnet"} 1
# HELP eth_con_spec_safe_slots_to_update_justified The number of slots to wait before updating the justified checkpoint.
# TYPE eth_con_spec_safe_slots_to_update_justified gauge
eth_con_spec_safe_slots_to_update_justified{module="spec",node="consensus"} 8
# HELP eth_con_spec_seconds_per_eth1_block The number of seconds per ETH1 block.
# TYPE eth_con_spec_seconds_per_eth1_block gauge
eth_con_spec_seconds_per_eth1_block{module="spec",node="consensus"} 14
# HELP eth_con_spec_seconds_per_slot The number of seconds per slot.
# TYPE eth_con_spec_seconds_per_slot gauge
eth_con_spec_seconds_per_slot{module="spec",node="consensus"} 12
# HELP eth_con_spec_slots_per_epoch The number of slots per epoch.
# TYPE eth_con_spec_slots_per_epoch gauge
eth_con_spec_slots_per_epoch{module="spec",node="consensus"} 32
# HELP eth_con_spec_sync_committee_size The sync committee size.
# TYPE eth_con_spec_sync_committee_size gauge
eth_con_spec_sync_committee_size{module="spec",node="consensus"} 512
# HELP eth_con_spec_target_committee_size The target committee size.
# TYPE eth_con_spec_target_committee_size gauge
eth_con_spec_target_committee_size{module="spec",node="consensus"} 128
# HELP eth_con_spec_terminal_block_hash_activation_epoch The epoch at which the terminal block hash is activated.
# TYPE eth_con_spec_terminal_block_hash_activation_epoch gauge
eth_con_spec_terminal_block_hash_activation_epoch{module="spec",node="consensus"} 1.8446744073709552e+19
# HELP eth_con_spec_terminal_total_difficulty The terminal total difficulty.
# TYPE eth_con_spec_terminal_total_difficulty gauge
eth_con_spec_terminal_total_difficulty{module="spec",node="consensus"} 1.079e+07
# HELP eth_con_spec_terminal_total_difficulty_trillions The terminal total difficulty in trillions.
# TYPE eth_con_spec_terminal_total_difficulty_trillions gauge
eth_con_spec_terminal_total_difficulty_trillions{module="spec",node="consensus"} 0
# HELP eth_con_sync_distance The sync distance of the node.
# TYPE eth_con_sync_distance gauge
eth_con_sync_distance{module="sync",node="consensus"} 2
# HELP eth_con_sync_estimated_highest_slot The estimated highest slot of the network.
# TYPE eth_con_sync_estimated_highest_slot gauge
eth_con_sync_estimated_highest_slot{module="sync",node="consensus"} 5.904653e+06
# HELP eth_con_sync_head_slot The current slot of the node.
# TYPE eth_con_sync_head_slot gauge
eth_con_sync_head_slot{module="sync",node="consensus"} 5.904651e+06
# HELP eth_con_sync_is_syncing 1 if the node is in syncing state.
# TYPE eth_con_sync_is_syncing gauge
eth_con_sync_is_syncing{module="sync",node="consensus"} 0
...
```

**Comment by samcm** [comment]
Thanks for the very detailed issue @jmc000! 

This one should be resolved with this change: https://github.com/ethpandaops/ethereum-metrics-exporter/pull/84. Just going to hold off for another upstream change to land before I merge it. That error should be safe to ignore for now.

Cheers!

**Comment by samcm** [comment]
Sorry this one slipped by me, should be resolved in `v0.23.0` - https://github.com/ethpandaops/ethereum-metrics-exporter/releases/tag/v0.23.0. Thanks for the issue!
